components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts

```diff
@@ -102,65 +197,109 @@ export class AutomateSettingsComponent implements OnInit {
+  // This prevents a user from being allowed to enter negative numbers
+  // or other actions that we dont want to allow
+  public preventNegatives(key: string) {
+    const allowedKeys = [
+      '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+      'Backspace', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown', 'Tab'
+    ];
+    return allowedKeys.includes(key);
   }
 
-  // patchDisableValue is the workaround for the chef-checkbox molecule since it is not
-  // an input annotation we need to patch the value inside the FormGroup
-  public patchDisableValue(form, checked: boolean) {
-    form.controls['disable'].setValue(checked);
+  public handleFormActivation(form, checked: boolean): void {
+    // patchValue does not mark the form dirty, so we need to do it manually;
+    form.get('disabled').markAsDirty();
+    // patchValue is a workaround for the chef-checkbox because we need to be
+    // able to store a reference to it being checked or not
+    form.patchValue({
+      disabled: !checked
+    });
+
+    // this loops through the rest of the form to disable the relevant controls;
+    const pertinentGroups = [ 'unit', 'threshold' ];
+    pertinentGroups.forEach(control => {
+      if ( checked ) {
+        form.get(control).enable();
+      } else {
+        form.get(control).disable();
+      }
+    });
   }
 
   // Apply the changes that the user updated in the forms
   public applyChanges() {
-    // (@afiune) At the moment the only two forms that are enabled are:
-    // => 'missingNodes'
-    // => 'deleteMissingNodes'
-    //
-    // We will apply the changes on the rest when we expose the forms
+    this.saving = true;
+    // Note: Services are currently not enabled through the form
     const jobs: IngestJob[] = [
-      IngestJobs.MissingNodes,
-      IngestJobs.MissingNodesForDeletion
+      // Event Feed
+      IngestJobs.EventFeedRemoveData,
+      IngestJobs.EventFeedServerActions,
+      // Service Groups
+      IngestJobs.ServiceGroupNoHealthChecks,
+      IngestJobs.ServiceGroupRemoveServices,
+      // Client Runs
+      IngestJobs.ClientRunsRemoveData,
+      IngestJobs.ClientRunsLabelMissing,
+      IngestJobs.ClientRunsRemoveNodes,
+      // Compliance
+      IngestJobs.ComplianceRemoveReports,
+      IngestJobs.ComplianceRemoveScans
     ].map(jobName => {
       const jobForm = this.getJobForm(jobName);
-      const job = new IngestJob(null);
-      job.name = jobName;
-      job.running = !jobForm.disable;
-      job.threshold = jobForm.threshold + jobForm.unit;
+      const isNested = jobForm.nested_name ? true : false;
+      const job = new IngestJob(null, null);
+      job.category = jobForm.category;
+      job.name = jobForm.name;
+      job.disabled = jobForm.disabled;
+
+      // If the user doesn't enter any number at all - this defaults to 0
+      if (jobForm.threshold === null) {
+        job.threshold = '0' + jobForm.unit;
+      } else {
+        job.threshold = jobForm.threshold + jobForm.unit;
```

Comment:
This is one of those cases that is needed because of the difference between the older pieces of the API and the newer pieces of the API.  

The older pieces of the API store/read the threshold value as a string with both the unit and threshold values.  For example, if the value is "4 days", the threshold on this portion of the api is stored as "4d".

The newer portions of the API are only storing the value for the amount of days to between retention.  and they are storing them as a number.  ie. numeric 4.

From how i'm looking at it, we can't really avoid packing and splitting.  But if theres some other way that i'm not seeing - i'm all ears.

If possibly you're suggesting simply renaming `job.threshold` to something like 'job.packed_threshold` so that ultimately that line would read as
`job.packed_threshold = jobForm.threshold + jobForm.unit` or something like that - maybe that would be easier to read?

Reply:
Roger that; sounds like you do have justification for the pack/split. ✔️ 

Your rename idea is probably a good thought, though. (IIRC you were using the variable `threshold` in some places to be the unpacked value and in others to be the packed value, which is a bit confusing.)